### DISCUSSION TITLE
Mocking/Spying/Stubbing Signals

### DISCUSSION THREAD

**Discussion by oravecz** [preactjs/signals#506] [General]
Is there a library or technique which can be used to mock signals?

When we string together an entire state flow using signals and effects, we make the testing of any method that relies on a signal's value extremely difficult because downstram notifications are going to fire.

It would be convenient if there was a method to set the signal's value without triggering notifications for these types of cases.

How do you test code with a heavy use of signals today?

**Comment by rschristian** [comment]
I suppose you can do the opposite, mock out computed and effects so that they don't track signal changes?

Would generally advise against mocks though -- testing anything other than the code you're actually running in production is of questionable value.
